### PR TITLE
Using one curator instance for leader election and data transfer

### PR DESCRIPTION
**Pull request by zen-dog**
Previously Marathon used two separate apache curator instances and hence two zookeeper connections - one for leader election and one for data transfer. This could have potential implications should leader client disconnect and the data client continues working and additionally wastes memory and threads. This commit merges both clients to one.

Fixes: MARATHON-8509

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala [hunk 1]
```diff
@@ -88,7 +88,7 @@ class CoreModuleImpl @Inject() (
     eventStream,
     hostPort,
     crashStrategy,
-    richCuratorFramework.client,
+    richCuratorFramework.client.usingNamespace(null), // using non-namespaced client for leader-election
```
Comment (timcharper) [inline src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala]: hmm, can we go the other way? Start with no namespace, and then add it for the storage parts?

Comment (zen-dog) [reply src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala]: I guess we could but should we? `/marathon/leader-curator` is the only exception - everything else *should* got into`/marathon/state`

Comment (timcharper) [reply src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala]: Personally, I'd prefer it, but I don't have an exact logical explanation as to why. I just find it a little surprising to see us set some namespace on the original connection, only to undo it later on for a different use case. There'd be less "overriding" or more "refining" if we started with a more general connection config, first.

File: src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala [hunk 2]
```diff
@@ -29,3 +29,12 @@ case object JvmExitsCrashStrategy extends CrashStrategy {
     Runtime.getRuntime.asyncExit(reason.code)(ExecutionContext.Implicits.global)
   }
 }
+
+/**
+  * For testing purposes only: This crash strategy will *NOT* crash but throw an exception
+  */
+case object ThrowExceptionAndDontCrashStrategy extends CrashStrategy {
```
Comment (timcharper) [inline src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: let's put this in src/test then?

Comment (timcharper) [reply src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: alongside src/test/scala/mesosphere/marathon/test/TestCrashStrategy.scala

File: src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala (hunk 1 above)
Comment (zen-dog) [reply src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala]: > set some namespace on the original connection, only to undo it later on for a different use case.

I'd say there is more potential to screw things up, by name-spacing (or rather forgetting) *every other* usage of the client *except* for leader-election. Again, a *namespace is the rule, no namespace is the exception*. But if you absolutely insist on it...

File: src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala (hunk 2 above)
Comment (zen-dog) [reply src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: Good point. Done!

**Pull request merged**